sql/core/src/main/scala/org/apache/spark/sql/json/JacksonGenerator.scala

```diff
@@ -74,4 +76,60 @@ private[sql] object JacksonGenerator {
 
     valWriter(rowSchema, row)
   }
+
+  /** Transforms a single InternalRow to JSON using Jackson
+   *
+   * TODO: make the code shared with the other apply method.
+   *
+   * @param rowSchema the schema object used for conversion
+   * @param gen a JsonGenerator object
+   * @param row The row to convert
+   */
+  def apply(rowSchema: StructType, gen: JsonGenerator, row: InternalRow): Unit = {
```

Comment:
The logic here is exactly the same with function `def apply(rowSchema: StructType, gen: JsonGenerator, row: Row)`, not sure if there any simple way to make the code shared.